content/tools-and-utilities/self-assignable-roles.md

```diff
@@ -0,0 +1,213 @@
++++
+title = 'Self Assignable Roles'
+weight = 2
++++
+
+## Tutorial Video
+
+{{< youtube IaTfJ4vqHhc >}}
+
+[Tutorial Video](https://www.youtube.com/watch?v=IaTfJ4vqHhc)
+
+## Roles
+
+{{% notice warning %}}
+
+Make sure that the bot has the _manage role_ permission and that the bot's role is **above** the role it is trying to assign.
+
+{{% /notice %}}
+
+{{% notice warning %}}
+
+If you want to use any of the `rolemenu` commands, you **need** to have the `MANAGE_GUILD` permission, or the Manage Server permission. This is hardcoded, meaning that command overrides will not affect it.
+
+{{% /notice %}}
+
+\
+Simply give the role command a name and then select which role you want the bot to assign to the person.&#x20;
+
+![Creation of a role command](role_command_creation.png)
+
+### Optional features
+
+{{% notice warning %}}
+
+Do **NOT** set the require role to the role you are assigning. You generally will not want to set the ignore role to the role you are assigning either _unless_ you wish to prevent the user from removing that role through the role-menu.
+
+{{% /notice %}}
+
+**Require role**: Requires a person to have a certain role in order for them to give themselves this role.\
+**Ignore role**: Ignore any person who has this role.\
+**Group:** Put the role in a role group (will be explained more further down).
+
+## Role Groups
+
+{{% notice info %}}
+
+Role groups are essential if you want to set up a role menu.
+
+{{% /notice %}}
+
+Role groups are useful for applying restrictions on a group of roles such as only being able to have one or the other role etc. They're also essential if you want to create a role menu. To create one, simply give the role group a name and then select which mode you want the role group to use
+
+Every role group, even Ungrouped has the option to delete all roles inside that group, other groups will not be affected. Be careful with that, it's permanent and can't be undone.
+
+![Creation of a role command group](role_command_group_creation.png)
+
+### The different modes
+
+1. None: This mode does nothing other than checking against the roles required and ignored roles.
+2. Single: They can only have 1 role in the group. (e.g. for colors)
+3. Multiple: You can set the minimum and maximum number of roles a member can have in the group.
+
+### Optional features
+
+{{% notice warning %}}
+
+Do **NOT** set the require role to the role you are assigning. You generally will not want to set the ignore role to the role you are assigning either _unless_ you wish to prevent the user from removing that role through the rolemenu.
```

Comment:
```suggestion
Do **not** set the require role to the role you are assigning. You generally will not want to set the ignore role to the role you are assigning either _unless_ you wish to prevent the user from removing that role through the rolemenu.
```